ui/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js

```diff
@@ -58,161 +39,34 @@ export default class ImportWithSeedPhrase extends PureComponent {
-        seedPhraseError = this.context.t('seedPhraseReq');
-      } else if (!isValidMnemonic(parsedSeedPhrase)) {
-        seedPhraseError = this.context.t('invalidSeedPhrase');
-      }
-    }
-
-    this.setState({ seedPhrase, seedPhraseError });
-  }
-
-  handlePasswordChange(password) {
-    const { t } = this.context;
-
-    this.setState((state) => {
-      const { confirmPassword } = state;
-      let confirmPasswordError = '';
-      let passwordError = '';
-
-      if (password && password.length < 8) {
-        passwordError = t('passwordNotLongEnough');
-      }
-
-      if (confirmPassword && password !== confirmPassword) {
-        confirmPasswordError = t('passwordsDontMatch');
-      }
-
-      return {
-        password,
-        passwordError,
-        confirmPasswordError,
-      };
-    });
-  }
-
-  handleConfirmPasswordChange(confirmPassword) {
-    const { t } = this.context;
-
-    this.setState((state) => {
-      const { password } = state;
-      let confirmPasswordError = '';
-
-      if (password !== confirmPassword) {
-        confirmPasswordError = t('passwordsDontMatch');
-      }
-
-      return {
-        confirmPassword,
-        confirmPasswordError,
-      };
-    });
-  }
-
-  handleImport = async (event) => {
-    event.preventDefault();
-
-    if (!this.isValid()) {
-      return;
-    }
-
-    const { password, seedPhrase } = this.state;
+  handleImport = async (password, seedPhrase) => {
     const {
       history,
       onSubmit,
       setSeedPhraseBackedUp,
       initializeThreeBox,
     } = this.props;
 
-    try {
-      await onSubmit(password, this.parseSeedPhrase(seedPhrase));
-      this.context.metricsEvent({
-        eventOpts: {
-          category: 'Onboarding',
-          action: 'Import Seed Phrase',
-          name: 'Import Complete',
-        },
-      });
-
-      setSeedPhraseBackedUp(true).then(async () => {
-        initializeThreeBox();
-        history.replace(INITIALIZE_END_OF_FLOW_ROUTE);
-      });
-    } catch (error) {
-      this.setState({ seedPhraseError: error.message });
-    }
-  };
-
-  isValid() {
-    const {
-      seedPhrase,
-      password,
-      confirmPassword,
-      passwordError,
-      confirmPasswordError,
-      seedPhraseError,
-    } = this.state;
-
-    if (
-      !password ||
-      !confirmPassword ||
-      !seedPhrase ||
-      password !== confirmPassword
-    ) {
-      return false;
-    }
-
-    if (password.length < 8) {
-      return false;
-    }
-
-    return !passwordError && !confirmPasswordError && !seedPhraseError;
-  }
-
-  onTermsKeyPress = ({ key }) => {
-    if (key === ' ' || key === 'Enter') {
-      this.toggleTermsCheck();
-    }
-  };
-
-  toggleTermsCheck = () => {
+    await onSubmit(password, seedPhrase);
     this.context.metricsEvent({
       eventOpts: {
         category: 'Onboarding',
         action: 'Import Seed Phrase',
-        name: 'Check ToS',
```

Comment:
This event was removed because it seems completely and utterly useless

Conceivably we could track the checking of the ToS and clicking on the privacy policy as fragments, to detect dropoff at this step. That might be useful. But as this event currently exists it's not useful at all, so I've just removed it for now.

The diff is misleading; the Import Complete event was already a thing, shown above as a removed line.

Reply:
This comment is outdated; this event is now preserved in the CreateNewVault component. It's useless but still better to keep this PR focused. Plus it is present in some reports in Mixpanel as well.